src/modules/land_detector/MulticopterLandDetector.cpp

```diff
@@ -174,7 +174,16 @@ bool MulticopterLandDetector::_get_ground_contact_state()
 			vertical_velocity_threshold *= 2.5f;
 		}
 
-		_vertical_movement = (fabsf(_vehicle_local_position.vz) > vertical_velocity_threshold);
+		if (_vehicle_local_position.v_z_valid && (fabsf(_vehicle_local_position.vz) < vertical_velocity_threshold)) {
+			_vertical_movement = false;
+
+		} else if (_vehicle_local_position.z_valid && (fabsf(_vehicle_local_position.z_deriv) < vertical_velocity_threshold)) {
```

Comment:
Could `z_deriv` ever be not valid in even just one sample even if `z_valid`?

Reply:
The condition could also be:
`(vz_valid && vz_low) || (z_valid && z_deriv_low)`
I wonder if that can ever give a false positive. I currently can't think of why it should other than the biased case. Maybe `z_deriv` is more filtered?

Reply:
> Could z_deriv ever be not valid in even just one sample even if z_valid?

I don't think so, it's also how it's used in the position controller